source/PlayerInfo.cpp

```diff
@@ -1011,6 +1012,42 @@ int PlayerInfo::ReorderShips(const set<int> &fromIndices, int toIndex)
 
 
 
+void PlayerInfo::ReorderShips(const vector<shared_ptr<Ship>> newOrder)
+{
+	// Check if the incoming vector contains the same elements
+	bool same = true;
+	size_t aSize = newOrder.size();
+	size_t bSize = ships.size();
+	
+	if(aSize != bSize)
+		same = false;
+	else
+	{
+		// Compare vectors element by element
+		for(size_t i = 0; i < aSize; ++i)
+		{
+			same = false;
+			for(size_t j = 0; i < bSize; ++j)
+			{
+				if(newOrder[i] == ships[j])
+				{
+					same = true;
+					break;
+				}
+			}
+			if(!same)
+				break;
+		}
+	}
+	
+	if(same)
+		ships = newOrder;
+	else
+		Files::LogError("Cannot reorder ships because the new order does not contain the same ships");
```

Comment:
This should be a fatal error.
```suggestion
		throw runtime_error("Cannot reorder ships because the new order does not contain the same ships");
```
This thrown error will then be caught by the main program loop and generate both an SDL pop-up and an entry in errors.txt.

(You'll need to `#include <stdexcept>` instead of Files.h